content/en/blog/_posts/2021-11-18-Call-to-Action-Survey-of-Kubernetes-Tool-Preferences-for-Programming-Tasks.md

```diff
@@ -0,0 +1,24 @@
+---
+layout: blog
+title: "Call to Action: Survey of Kubernetes Tool Preferences for Programming Tasks"
+date: 2021-11-18
+slug: kubernetes-crud-debugging-monitoring-tool-survey
+---
+
+**Authors:** Cora Coleman (UCSD) and Nick Mitchell (IBM)
+
+Understanding the software tools that are preferred by Kubernetes developers and whether preference depends on the programming task at hand, is important for informing the future design and development of these tools. To help understand how people with different levels of development experience are using and prefer to use Kubernetes, we are currently performing a 2-minute [Kubernetes Tasks and Tools survey](https://forms.gle/CcXwjx1Tftkp7KAW9) that you can still take! The survey asks whether you are a beginner, intermediate, or expert Kubernetes developer and for your preference among three tool modalities: Command Line Interface (CLI), Graphical Console, and IDE, to examine whether your tool preference varies across three programming task types: CRUD, Debugging, and Monitoring.
```

Comment:
questions in the form might need some review as well. Some examples of suggestions I have from the first glance:

- how k8s crud operations might be done via Grafana?
- how VScode can be used for debugging tasks? Do you refer to specific plugins?
- Monitoring of checking that app was created and it's CPU utilization are two very different tasks. CLI is easy to use for checking the status, while dashboards are the natural choice for metrics.
- Also list of clouds is a bit strange. Is OpenShift a cloud? why Google is not mentioned and IBM does? Would the survey benefit from comparing different K8s offering from AWS instead of making them into a single bullet point?

Reply:
I agree with Sergey on all points. The openshift is not a cloud provider, its a distro, gke isn't on there.

How is the data going to be used after outside of making it available?

FWIW, I am not okay with using k8s properties to advertise this more until we know more.

Reply:
Thank you for your important feedback on the survey questions @SergeyKanzhelev and @mrbobbytables. Currently we have 54 responses and we would hate to invalidate that data by changing the questions for this version. We will definitely mention those threats to validity that you each have pointed out (for example, in providing correct distinctions between distros and providers for the cloud provider question) in final results once the survey is closed for responses. However, future iterations of this survey should implement those changes and again, we appreciate your help in improving these questions! What are your thoughts in terms of the tradeoffs of rewriting the survey and invalidating current data versus publishing the blog with the survey as-is? 

With respect to how the data is going to be used after making it available, I was thinking this work could build on the [Apps SIG 2018 survey](https://docs.google.com/presentation/d/1utT0K-u1nl2apXRo29GaBvRV1x7mFLeQSgpw8mI_nGM/edit#slide=id.p) and present the findings in a similar fashion (as a living slides document or something similar). What are your thoughts on making a living document similar to this survey? I’d appreciate any feedback on what formats of our results would be useful for the community.

Reply:
Thank you for this information @jberkus. I would appreciate help with revising the survey to satisfy those requirements. How can I reach out to the ContribEx survey team for help?

> Please do not publish this survey on the Kubernetes blog.
> 
> No survey should be published in a Kubernetes community property without satisfying GDPR requirements, which this survey does not. Further, if it's going to be a survey "belonging" to a SIG, the survey document itself should be under SIG ownership.
> 
> The ContribEx survey team is available to help with revising the survey to support these requirements.

I spoke more with @mrbobbytables today on this and now understand that there is an [issue template](https://github.com/kubernetes/community/issues/new?assignees=&labels=area%2Fcontributor-comms%2C+sig%2Fcontributor-experience&template=survey-request.md&title=SURVEY+REQUEST%3A+%3Ctopic%3E) that I can submit to get our community survey questions reviewed. I will do this and get help with rewriting the survey.